### PR TITLE
Fix empty id management and message type

### PR DESCRIPTION
**Pull request by oliverpool**
I've split the test into a `sse_test` subpackage: do you have any objection about it?

### CODE REVIEW

**Comment by oliverpool** [comment]
@jadr2ddude I'm done with this PR.

I would really appreciate if you could review it :-)

File: scanner_test.go [hunk 1]
```diff
@@ -1,16 +1,18 @@
-package sse
+package sse_test
```
Comment (niaow) [inline scanner_test.go]: Please stop doing this. Your tests should be in the same package.

File: client_test.go [hunk 2]
```diff
@@ -1,16 +1,18 @@
-package sse
```
Comment (niaow) [inline client_test.go]: Same as the other one, should be sse

**Review by niaow — CHANGES_REQUESTED** [review]
Please put all tests in the SSE package

File: scanner_test.go (hunk 1 above)
Comment (oliverpool) [reply scanner_test.go]: @jadr2ddude why should the tests be in the same package? (I thought that is was better to use a `*_test` package)

**Pull request merged**